### ISSUE TITLE
"Private members are unused" fires on VB static holder types

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#30882]
To mimic C# static types in VB, it's common do the following (look up FxCop for StaticHolderTypes classification):

#### Repro steps

``` VB
    Friend NotInheritable Class SettingsValueSerializer

        Private Sub SettingsValueSerializer()
        End Sub
	End Class

```

#### Expected behavior
Not to fire

#### Actual behavior
Severity	Code	Description	Project	Project Rank	File	Line	Category	Suppression State	Tool
Message	IDE0051	Private member 'SettingsValueSerializer.SettingsValueSerializer' is unused.	Microsoft.VisualStudio.Editors	3	E:\project-system2\src\Microsoft.VisualStudio.Editors\SettingsDesigner\SettingsValueSerializer.vb	16	Code Quality	Active	Compiler

**Comment by mavasani** [comment]
I'd thought we already handled this case, we also have a test: http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests/RemoveUnusedMembers/RemoveUnusedMembersTests.vb,161

Will investigate.

**Comment by mavasani** [comment]
Ah, this is not a constructor being defined, but an unused private method with same name as the type. This looks most definitely to be an unused method that should be flagged. Likely intention was to define a private constructor `Private Sub New`, which would make it a static holder type, which is correctly detected and not flagged by us.

**Comment by mavasani** [comment]
Valid code with no compiler errors, we correctly flag unused method `SettingsValueSerializer`:
```
Friend NotInheritable Class SettingsValueSerializer
    Private Sub SettingsValueSerializer()
    End Sub
End Class

Class Class2
    Private field As New SettingsValueSerializer()
End Class
```

Compiler error on attempting to instantiate a static holder type, we do not flag the constructor as unused:
```
Friend NotInheritable Class SettingsValueSerializer
    Private Sub New()
    End Sub
End Class

Class Class2
    Private field As New SettingsValueSerializer()
End Class
```
